### PR TITLE
fix the test with fake.ClientSet

### LINKED ISSUE #8: Remove FakeAPIUtil
Migrating from https://github.com/kubernetes-incubator/external-storage/issues/786
/kind cleanup

### PR DESCRIPTION
**Pull request by fatsheep9146**
Fix #8

### CODE REVIEW

**Comment by fatsheep9146** [comment]
/cc @cofyc 

I modify the test code for deleter as an example to make sure whether I change the code as you expected.
Please review my code and let me know if somewhere is wrong, I will fix it again.

File: provisioner/pkg/deleter/deleter_test.go [hunk 1]
```diff
@@ -635,3 +645,86 @@ func waitForAsyncToComplete(t *testing.T, d *Deleter, pvNames ...string) {
...
+	fakeClient.AddReactor("create", "persistentvolumes", func(action core.Action) (handled bool, ret runtime.Object, err error) {
+		if track.shouldFail {
+			return false, nil, fmt.Errorf("API failed")
+		}
+		createAction := action.(core.CreateAction)
+		pv := createAction.GetObject().(*v1.PersistentVolume)
+		track.createdPVs[pv.Name] = pv
+		track.cache.AddPV(pv)
+		return true, createAction.GetObject(), nil
+	})
```
[the remark this answers is not in the document]
Comment (fatsheep9146) [reply provisioner/pkg/deleter/deleter_test.go]: Ok, I will try to check if the NewSimpleClient satisfies our demands.

**Comment by fatsheep9146** [comment]
/cc @cofyc 

I have already fixed the test with NewSimpleClient, please check it out, thx.

File: provisioner/pkg/deleter/deleter_test.go [hunk 2]
```diff
@@ -506,7 +515,30 @@ func testSetupForJobCleaning(t *testing.T, config *testConfig, cleanupCmd []stri
...
+			config.cache.DeletePV(pvName)
+			return false, nil, nil
+		}
+		return true, nil, errors.NewNotFound(v1.Resource("persistentvolumes"), pvName)
+	})
+
+	config.apiUtil = util.NewAPIUtil(config.clientset)
 	config.procTable = NewFakeProcTable()
 	config.jobControl = NewFakeJobController()
 	config.volUtil = util.NewFakeVolumeUtil(config.volDeleteShouldFail, map[string][]*util.FakeDirEntry{})
```
Comment (cofyc) [inline provisioner/pkg/deleter/deleter_test.go]: All `FakeVolumeUtil` related code must be replaced.

**Comment by fatsheep9146** [comment]
/cc @cofyc 
All tests using old FakeAPIUtil is modified and the FakeAPIUtil related codes are cleared.

File: provisioner/pkg/discovery/discovery_test.go [hunk 3]
```diff
@@ -668,3 +698,19 @@ func TestUseAlphaAPI(t *testing.T) {
...
+
+func getAndResetCreatedPVs(cli *fake.Clientset, cache *cache.VolumeCache) map[string]*v1.PersistentVolume {
+	pvs := make(map[string]*v1.PersistentVolume)
+	for _, action := range cli.Actions() {
+		if action.Matches("create", "persistentvolumes") {
+			obj := action.(core.CreateAction).GetObject()
+			pv := obj.(*v1.PersistentVolume)
+			if _, exists := cache.GetPV(pv.Name); exists {
+				pvs[pv.Name] = pv
+				// fmt.Printf("catch one create action pv name %s\n", pv.Name)
```
[the remark this answers is not in the document]
Comment (fatsheep9146) [reply provisioner/pkg/discovery/discovery_test.go]: ok

**Comment by cofyc** [comment]
Remove '[WIP]' when it is ready.

**Comment by fatsheep9146** [comment]
/cc @cofyc 
Done

**Pull request merged**